akka-projection-core/src/main/scala/akka/projection/StatusObserver.scala

```diff
@@ -31,10 +31,10 @@ abstract class StatusObserver[-Envelope] {
   def stopped(projectionId: ProjectionId): Unit
 
   /**
-   * Called when the envelope has been processed and corresponding offset stored.
+   * Called when the corresponding offset has been stored (even if the envelope has not been processed yet).
```

Comment:
```suggestion
   * Called when the corresponding offset has been stored.
```

I don't think we need to add that extra sentence. The method is solely about the progress of the offset. Since we have at-least-once, at-most-once and exactly-once, we will not when that thing happens.